src/cpp/server/server_context.cc

```diff
@@ -173,6 +173,14 @@ void ServerContext::AddTrailingMetadata(const grpc::string& key,
   trailing_metadata_.insert(std::make_pair(key, value));
 }
 
+void ServerContext::TryCancel() const {
+  grpc_call_error err = grpc_call_cancel_with_status(
+      call_, GRPC_STATUS_CANCELLED, "Cancelled on the server side", NULL);
+  if (err != GRPC_CALL_OK) {
```

Comment:
I think currently grpc_call_cancel_with_status never returns anything other than ok. I think this should be either removed or changed to at least a GPR_ERROR?